### DISCUSSION TITLE
How to apply voltage

### DISCUSSION THREAD

**Discussion by gutsfromberserk1** [pybamm-team/PyBaMM#4149] [Q&A]
Hey guys, 
Hope everyone is doing fine. I want to study cyclic voltammetry. I want to work on both batteries (using Chen2020) and crate a one electrode setup (getting rid of the separator in the example code for creating a model part). Therefore, I want to ask how can I apply voltage? Especially with a defined sweep rate? and also create a one electrode model? 

Thanks a lot,

**Comment by TomTranter** [comment]
Have a look at the docs please, these two in particular.
https://docs.pybamm.org/en/latest/source/examples/notebooks/simulations_and_experiments/custom-experiments.html#Custom-steps
https://docs.pybamm.org/en/stable/source/examples/notebooks/models/half-cell.html#Half-cell-models-in-PyBaMM

**Reply by gutsfromberserk1** [reply]
Hi, thanks for the fast reply.
I have tried this code: 
``` 
import pybamm
import numpy as np
import matplotlib.pyplot as plt 
model = pybamm.lithium_ion.DFN({"working electrode": "positive"})
params = pybamm.ParameterValues("Xu2019")

v_init = -0.3  # initial voltage
v_final = 0.3  # final voltage
scan_rate = 150e-3 # V/s 
num_periods = 10  
num_points_per_period = 100
total_time_per_period = (v_final - v_init) / scan_rate
time_per_period = np.linspace(0, total_time_per_period, num_points_per_period)


voltage_per_period = np.concatenate((np.linspace(v_init, v_final, num_points_per_period // 2),
                                     np.linspace(v_final, v_init, num_points_per_period // 2)))


voltage = np.tile(voltage_per_period, num_periods)
time = np.concatenate([time_per_period + i * total_time_per_period for i in range(num_periods)])

new_array = np.column_stack((time, voltage))


step = pybamm.step.voltage(value=new_array)
sol = pybamm.Simulation(
    model, experiment=step, parameter_values=params
)
sol.solve()
```
but get an error "OverflowError: cannot convert float infinity to integer" . Also, in the docs it says voltage should be positive but I want to start scanning from -0.3V, is it possible? (This error also pops up when I changed the scan to start 1V and end 2V. Therefore, I believe it is not related with the negative voltage value.)

Thanks,

**Reply by gutsfromberserk1** [reply]
Hi again, when I tried the example in the docs, it says "module 'pybamm.experiment.step' has no attribute 'CustomStepImplicit'" where I use pybamm version 24.1.

Thanks,

**Reply by agriyakhetarpal** [reply]
@gutsfromberserk1, I think you might be trying the example in the `latest` version of the documentation, rather than the `stable` version. https://github.com/pybamm-team/PyBaMM/pull/3835 added this functionality, and it will be available in the latest release soon (v24.5).
